### PR TITLE
Create README.md

### PR DESCRIPTION
**Pull request by Wenting0905**
PTAL @andelf @shenli

### CODE REVIEW

**Comment by andelf** [comment]
LGTM

File: sql/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# TiDB SQL Grammar
+
+TiDB supports the SQL-92 standard and is compatible with MySQL grammar. Currently, TiDB supports the majority of frequently-used MySQL grammars.
+
+## TOC
+
++ [TiDB SQL Syntax Diagram](https://pingcap.github.io/sqlgram/)
++ Language structure
+    - literal value
```
Comment (shenli) [inline sql/README.md]: value -> values

File: sql/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - database, table, index, column and alias
+    - keyword and reserved word
+    - user-defined variable
+    - expression
+    - annotation
++ Data type
```
Comment (shenli) [inline sql/README.md]: type -> types

**Comment by Wenting0905** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**